generator/namespaces.py

```diff
@@ -400,22 +401,49 @@ def namespaces(
 
     _merge_namespaces(namespaces, _get_metric_hub_namespaces(namespaces))
 
-    disallowed_namespaces = yaml.safe_load(disallowlist.read()) or {}
-    disallowed_regex = [
-        fnmatch.translate(namespace) for namespace in disallowed_namespaces
-    ]
-    disallowed_namespaces_pattern = re.compile("|".join(disallowed_regex))
-
-    updated_namespaces = {}
-    for namespace, _ in namespaces.items():
-        if (
-            not disallowed_namespaces_pattern.fullmatch(namespace)
-            and namespace not in ignore
-        ):
-            if "spoke" not in namespaces[namespace]:
-                namespaces[namespace]["spoke"] = DEFAULT_SPOKE
-            if "glean_app" not in namespaces[namespace]:
-                namespaces[namespace]["glean_app"] = False
-            updated_namespaces[namespace] = namespaces[namespace]
+    updated_namespaces = _filter_disallowed(namespaces, disallowlist)
+    for namespace, _ in updated_namespaces.items():
+        if namespace not in ignore:
+            if "spoke" not in updated_namespaces[namespace]:
+                updated_namespaces[namespace]["spoke"] = DEFAULT_SPOKE
+            if "glean_app" not in updated_namespaces[namespace]:
+                updated_namespaces[namespace]["glean_app"] = False
 
     Path("namespaces.yaml").write_text(yaml.safe_dump(updated_namespaces))
+
+
+def _filter_disallowed(namespaces, disallowlist):
+    """Filter models, explores and views from the generated namespaces config, based on the disallowlist."""
+
+    def match_any(patterns, name):
+        return any(fnmatch.fnmatch(name, p) for p in patterns)
+
+    # transform namespace disallowlist to a dict
+    disallowed_namespaces = yaml.safe_load(disallowlist.read()) or {}
```

Comment:
If the YAML gets loaded the top-level object is a list, so I'd have the fallback value be a list as well.
```suggestion
    disallowed_namespaces = yaml.safe_load(disallowlist.read()) or []
```